openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -145,7 +146,14 @@ def post(self, request):
         if not hasattr(request.user, 'backend'):
             request.user.backend = self._get_path_of_arbitrary_backend_for_user(request.user)
 
-        if not self._is_grant_password(request.auth):
+        access_token = request.auth
+        is_jwt_authenticated = isinstance(request.successful_authenticator, JwtAuthentication)
+        if is_jwt_authenticated:
```

Comment:
1. I'm not sure this is enough. JwtAuthentication allows both header and cookie auth, and I _think_ cookie auth is also a fall back? So, someone could supply a JWT token in the header that gets the right access token from DB, but isn't actually authenticated? I may need to talk to @timmc-edx about this.
2. Nit: I find the following easier to read, using the new [walrus operator](https://realpython.com/python-walrus-operator/).
```suggestion
        if is_jwt_authenticated := isinstance(request.successful_authenticator, JwtAuthentication):
```